src/js/mixins/stateful.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * @file mixins/stateful.js
+ */
+import * as Fn from '../utils/fn';
+import log from '../utils/log';
+import * as Obj from '../utils/obj';
+
+/**
+ * Set the state of an object by mutating its `state` object in place.
+ *
+ * @param  {Object|Function} next
```

Comment:
Changing to `stateUpdates` (`newState` seems to imply that it replaces `state`, which it doesn't).